### PR TITLE
Persistence Configs for Run 7

### PR DESCRIPTION
**Pull request by jbanovetz**
Three different Persistence Tests for Run 7. I wasn't able to run Seth's timing code, though I am also not sure if it is set up for the new `DARKINTERRUPT` or `DARKINTERRUPTLIST`, which I rely on for these config files.

### CODE REVIEW

File: Run7/step05/Persistence_large.cfg [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+#linearity # needs flat
+#fullwell  # needs flat
+#cti       # needs sflat
+#tearing   # needs sflat
+#persistence # needs persistence
+#scan      # needs scan
+
+[ANALYSIS_RUNS]
+# if only doing a portion of the ACQUIRE, use prior run for GAIN, DARK, BADPIXEL
+#GAIN=curated_amp_gains_2020-09-16.json         # get GAIN from this file
```
Comment (jchiang87) [inline Run7/step05/Persistence_large.cfg]: I think you can just delete the `[ANALYZE]` and `[ANALYSIS_RUNS]` sections.

File: Run7/step05/Persistence_large.cfg [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+[FLAT]
+DARKINTERRUPT = True # If True, will take darks between flat images 
+DARKINTERRUPTLIST = 15  8 # integration time and image count for dark set, formatted the same as the DARK acquisition
+SHUTTER = OPEN  # Shutter opens at start of acqusition, closes at end
+BCOUNT=0        # number of bias frames per flat image
+WL    =red    # wavelength filter to use for the flats
+HILIM =15.0    # maximum seconds for a flat field exposure
+LOLIM =0        # minimum seconds for a flat field exposure
+FILTERCONFIG = /gpfs/slac/lsst/fs2/u1/devel/ccs/bot-eotest-configs/Run6/calib.cfg
```
Comment (jchiang87) [inline Run7/step05/Persistence_large.cfg]: Please update this path to 
```
FILTERCONFIG = /home/ccs/bot-eotest-configs/Run7/calib.cfg
```

File: Run7/step05/Persistence_large.cfg [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+FLAT=15     90000,  # number of electrons/pixel, ND filter
```
Comment (jchiang87) [inline Run7/step05/Persistence_large.cfg]: This comment, `# number of electrons/pixel, ND filter`, looks wrong.

File: Run7/step05/Persistence_large.cfg [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+     15     119000,
+     15     120000,
+     15     130000,
+     15     140000,
+     15     150000,
+     15     160000,
+     15     170000,
+     15     180000,
+     15     190000,
+     15     200000,
```
Comment (jchiang87) [inline Run7/step05/Persistence_large.cfg]: Can you add a final `\n` (newline) to this file?

File: Run7/step05/Persistence_medium.cfg [hunk 5]
```diff
@@ -0,0 +1,32 @@
+[ACQUIRE]
+sflat
+
+[SFLAT]
+DARKINTERRUPT = True # If True, will take darks between flat images 
+DARKINTERRUPTLIST = 15  5 # integration time and image count for dark set, formatted the same as the DARK acquisition
+SHUTTER = OPEN  # Shutter opens at start of acqusition, closes at end
+FILTERCONFIG = /gpfs/slac/lsst/fs2/u1/devel/ccs/bot-eotest-configs/Run6/calib.cfg
```
Comment (jchiang87) [inline Run7/step05/Persistence_medium.cfg]: This needs to be updated.

File: Run7/step05/Persistence_short.cfg [hunk 6]
```diff
@@ -0,0 +1,21 @@
+[ACQUIRE]
+sflat
+
+
+[SFLAT]
+DARKINTERRUPT = True # If True, will take darks between flat images 
+DARKINTERRUPTLIST = 15  5 # integration time and image count for dark set, formatted the same as the DARK acquisition
+SHUTTER = OPEN  # Shutter opens at start of acqusition, closes at end
+FILTERCONFIG = /gpfs/slac/lsst/fs2/u1/devel/ccs/bot-eotest-configs/Run6/calib.cfg
```
Comment (jchiang87) [inline Run7/step05/Persistence_short.cfg]: Please update.

File: Run7/step05/Persistence_short.cfg [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+LOLIM =0       # minimum seconds for a flat field exposure
+SFLAT=red  80000     1,   # wavelength filter, signal(e-), count
+      red  85000    1,    # wavelength filter, signal(e-), count
+      red  90000    1,    # wavelength filter, signal(e-), count
+      red  95000    1,    # wavelength filter, signal(e-), count
+      red  100000    1,    # wavelength filter, signal(e-), count
+      red  105000    1,    # wavelength filter, signal(e-), count
+      red  110000    1,    # wavelength filter, signal(e-), count
+      red  115000    1,    # wavelength filter, signal(e-), count
+      red  120000    1,    # wavelength filter, signal(e-), count
```
Comment (jchiang87) [inline Run7/step05/Persistence_short.cfg]: Please add a newline.

**Review by jchiang87 — CHANGES_REQUESTED** [review]
The github actions CI check runs Seth's script as a syntax checker, so you don't need to do that.

I requested a number of changes, the key one being to update the `FILTERCONFIG` values in all three files.

I'm curious why a `flat` acquisition is used for `Persistence_large.cfg`, but `sflat` for the other two.

**Comment by jbanovetz** [comment]
Thank you for looking at it, I will make those changes.

> I'm curious why a `flat` acquisition is used for `Persistence_large.cfg`, but `sflat` for the other two.

For this one, I wanted two flats taken at the same flux level so that I could average them to beat down any read or shot noise. I could change it to sflats and just copy every flux level X number of times (maybe 3) if that would help with processing.

**Review by jchiang87 — APPROVED** [review]
lgtm.

**Comment by jbanovetz** [comment]
Looking at the output of Seth's code I don't think that it is picking up the `DARKINTERRUPT` command as the `Persistence_large.cfg` is only taking ~25 minutes right now and it should have 8, 15 second darks between flat images (rough estimate would be around 205 mintues). I think the configs are good to go but we might want to change that timing script at some point.

**Comment by jchiang87** [comment]
sure, I don't think the point was to get the timing results though, but rather to do simple syntax checking of the file.

**Comment by seanmacb** [comment]
[Reference for changes](https://rubinobs.atlassian.net/wiki/spaces/CAM/pages/60358677)

**Comment by jbanovetz** [comment]
After some discussion in Aaron's weekly meeting, I changed the Large and Short configs to match what he was thinking. This included changing the Large config to `sflats` and it should be about 4 hours.

**Comment by sethdigel** [comment]
I'd recommend removing the comma at the end of the last line of lists of flats/sflats.  I recall that fp_srcipts does not like it.

**Comment by jbanovetz** [comment]
Removed the commas and will now merge

**Pull request merged**